src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1690,17 +1600,15 @@ MavlinkReceiver::handle_message_play_tune(mavlink_message_t *msg)
 
 	char *tune = play_tune.tune;
 
-	if ((mavlink_system.sysid == play_tune.target_system ||
-	     play_tune.target_system == 0) &&
-	    (mavlink_system.compid == play_tune.target_component ||
-	     play_tune.target_component == 0)) {
+	if ((mavlink_system.sysid == play_tune.target_system || play_tune.target_system == 0) &&
+	    (mavlink_system.compid == play_tune.target_component || play_tune.target_component == 0)) {
 
 		if (*tune == 'M') {
-			int fd = px4_open(TONEALARM0_DEVICE_PATH, PX4_F_WRONLY);
+			if (_tune_control_sub == nullptr) {
+				_tune_control_sub = orb_advertise(ORB_ID(tune_control), &tune);
 
-			if (fd >= 0) {
-				px4_write(fd, tune, strlen(tune) + 1);
-				px4_close(fd);
+			} else {
+				orb_publish(ORB_ID(tune_control), _tune_control_sub, &tune);
```

Comment:
This looks wrong? https://github.com/PX4/Firmware/blob/master/msg/tune_control.msg

Reply:
It might be... It behaves the same as previous, which means the previous is probably also broken... I've been digging into this one, David and I discussed it earlier today. Any advice?

Reply:
The main issue is that tune_control is already broken for `M` args...

Reply:
@dagar I'm planning on working more on this after #11194.

Reply:
@dagar, @mcsauder - It was broken since tune control came in. Passing a string to a none serviced IOCTL. The first decision is: Do we need it? If not removed it and return and error. If so there are 2 choices. 

1. Simple: Add static member in the tune classes that owns the compiled tune desc file to look up the ascii string in the tone table and return its ordinal. It it returns -1 eissues an error and exit. 

2. Complex: Add code with resource protection in the tune lib to play an arbitrary tune string like as done in the tune_control app but non blocking.

My preference is #1 with a fall back that if in the future someone says it is broken, we can always do #2 but that is based on the features value and my work load. On the other hand if someone has time wants to learn, this is a good little project.

Reply:
@dagar, I reverted this section to help this PR move forward.  @davids5 , I'll get the IOCTL work finished up in a follow-up PR.